### PR TITLE
Setting up CI

### PR DESCRIPTION
**Pull request by nmittler**
Copied from quinn.

### CODE REVIEW

File: .github/workflows/codecov.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - uses: actions-rs/tarpaulin@v0.1
+      - uses: codecov/codecov-action@v1.0.2
+        #with:
```
Comment (nmittler) [inline .github/workflows/codecov.yml]: @djc IIUC the token is only needed for private repos?

Comment (djc) [reply .github/workflows/codecov.yml]: Yup, should not longer be needed. But also see https://github.com/quinn-rs/quinn/pull/1505.

File: .github/workflows/codecov.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - uses: taiki-e/install-action@cargo-llvm-cov
+      - run: cargo llvm-cov --all-features --workspace --lcov --output-path lcov.info
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          files: lcov.info
```
Comment (nmittler) [inline .github/workflows/codecov.yml:22]: @djc is there a file needed somewhere in the repo? I didn't see any files added in https://github.com/quinn-rs/quinn/pull/1505.

File: .github/workflows/codecov.yml (hunk 1 above)
Comment (nmittler) [reply .github/workflows/codecov.yml]: Thanks! Updated to the new codecov.

File: .github/workflows/codecov.yml (hunk 2 above)
Comment (djc) [reply .github/workflows/codecov.yml:22]: No, this is the output from `cargo llvm-cov`, as seen in the `--output-path` above.

File: .github/workflows/codecov.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+          fail_ci_if_error: true
```
Comment (djc) [inline .github/workflows/codecov.yml]: Maybe add the trailing newline here?

**Comment by nmittler** [comment]
@djc PTAL

File: Cargo.toml [hunk 4]
```diff
@@ -9,4 +9,7 @@ categories = ["network-programming", "asynchronous"]
 edition = "2021"
 rust-version = "1.59"
 
+[badges]
```
Comment (djc) [inline Cargo.toml:12]: IIRC crates.io doesn't even show these anymore, so there's little value in having them.

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
FWIW, I've disabled the branch protection so you should be able to merge PRs without any review and/or push directly to main. Not sure why that was set up the way it was...

File: Cargo.toml (hunk 4 above)
Comment (nmittler) [reply Cargo.toml:12]: Ah good to know.

**Comment by nmittler** [comment]
> FWIW, I've disabled the branch protection so you should be able to merge PRs without any review and/or push directly to main. Not sure why that was set up the way it was...

Ah I had enabled it ... but it's fine either way.

**Pull request merged**